furnace.c

```diff
@@ -333,6 +399,56 @@ do_pwm_work(furnace_context_t *ctx)
   gpio_put(FURNACE_FIRE_PIN, ctx->pwm_current);
 }
 
+int
+sgn(int val_1, int val_2)
+{
+  if(val_1 > val_2)
+    return -1;
+  return 1;
+}
+
+static inline uint8_t
+clamp_val(int min, int max, int val)
```

Comment:
Consider naming it `clamp_u8` since the returned type is `uint8_t` and I think argument types should be the same as return type. Otherwise user might also get some truncation because of narrow type conversion, thus unexpected behavior.